### PR TITLE
Newsletter Web Page and Base Features

### PR DESCRIPTION
**Pull request by mngshm**
This is a base PR for the FOSS United Newsletter on new Platform. PR Includes: 
- Newsletter UI (subject to change) 
- Email Group for FOSS United Newsletters
- Subscribe to newsletter feature with frappe.call
- Temporary success page (subject to change - mostly in the next PR)

Demonstration: 
https://github.com/fossunited/fossunited/assets/82205152/c582a20d-765b-4c06-b997-7b6b67ab49ef

### CODE REVIEW

**Comment by harshtandiya** [comment]
Attach screenshots. The demo is not working

**Comment by mngshm** [comment]
image

image

**Comment by RiyaMathew-11** [comment]
Looks clean. Maybe instead of the dummy text in the email address field, keep a dummy placeholder email - maybe username@example.com. 

We might need to merge design styles too.

**Review by RiyaMathew-11 — COMMENTED** [review]
Looks clean. Maybe instead of the dummy text in the email address field, keep a dummy placeholder email - maybe [username@example.com](mailto:username@example.com).

We might need to merge design styles too.

**Comment by mngshm** [comment]
Sure, I'll push an update.

**Comment by harshtandiya** [comment]
Can you put a screenshot of success page too

**Comment by mngshm** [comment]
image

**Comment by harshtandiya** [comment]
Make it a bit better with a green tick maybe

**Comment by mngshm** [comment]
Cheese_Wed-13Dec23_18 36

File: fossunited/www/newsletter/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+function subscribeHandler(event) {
+  var emailInput = document.getElementById("email").value;
```
Comment (harshtandiya) [inline fossunited/www/newsletter/index.js]: use let instead of var as much as possible

File: fossunited/www/newsletter/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+function showSuccessMsg() {
+  setTimeout( function (){
+    document.getElementById("subscribed-page").style.display = 'block';
+  }, 500);
+
+  var sections = ["header", "newsletters-list", "footer-cta", "subscribe-section"];
+  for (var i=0; i < sections.length; i++) {
+    document.getElementById(sections[i]).style.display = 'none';
+  }
```
Comment (harshtandiya) [inline fossunited/www/newsletter/index.js]: let's replace var with let

File: fossunited/www/newsletter/index.html [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+  <div class="base-container container" style="max-width: 1000px">
+    <h4 class="text-center">Past Newsletters</h4>
+    <div>
+      <ul class="m-auto py-8 list-unstyled gap-2 text-left row justify-content-center">
+        {% for newsletter in newsletters %}
+        <li id="news" class="py-4 text-center col-md-3 position-relative">
+          {{ newsletter.subject }}
+          <a class="stretched-link" href="{{ frappe.get_url() }}/{{ newsletter.route }}"></a>
+        </li>
+        {% endfor %}
```
Comment (harshtandiya) [inline fossunited/www/newsletter/index.html]: Kind of redundant. I suggest replacing the `<li>` with `<a>` and remove line 41.

File: fossunited/www/newsletter/index.html [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+  <div id="cta-container" class="base-container container h-full">
+    <div id="cta">
+      <div class="text-center">
+        <h4>Connect with us</h4>
+        <p>Drop a message in our telegram group.</p>
+        <div>
+          <button type="submit" id="telegram-channel">
+            <a href="https://t.me/fossunited" alt="@fossunited telegram">Get in touch</a>
+          </button>
+        </div>
```
Comment (harshtandiya) [inline fossunited/www/newsletter/index.html:56]: being nitpicky but the outer div is not needed I guess

**Review by harshtandiya — CHANGES_REQUESTED** [review]

**Review by harshtandiya — APPROVED** [review]

**Pull request merged**